src/common/catalog-entities/kubernetes-cluster.ts

```diff
@@ -97,6 +97,20 @@ export class KubernetesClusterCategory extends EventEmitter implements CatalogCa
     }
   };
 
+  constructor() {
+    super();
+
+    this.on("onCatalogAddMenu", (ctx: CatalogEntityAddMenuContext) => {
+      ctx.menuItems.push({
+        icon: "text_snippet",
+        title: "Add from kubeconfig",
+        onClick: async () => {
+          ctx.navigate("/add-cluster");
+        }
+      });
+    });
+  }
```

Comment:
I think it would be a better design (and would prevent malicious extensions) if this function had to return the items it wished to add, and we do the combining ourselves.

Reply:
I think the current model already supports "sandboxing" of menuItems (if we want to implement it). Basically we could give an empty menuItems to each callback and merge them together.

Reply:
True but from an API perspective (imo) it feels more intuitive to return what I want to add instead of modifying my own input.

Reply:
I would like to get more feedback on this. How about if we keep current implementation and change it based on feedback (if needed, in follow-up PR)?